### PR TITLE
Create Ali_Project4

### PR DESCRIPTION
**Pull request by ahasanzadeh13**
@giacomodecolle @peihongx could you please look at the questions I have made so far? Thanks.

### CODE REVIEW

File: Project 4/Ali_Project4 [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Ali’s Kata 7 question: SELECT queries are used for selecting bindings for variables, but you can also ask YES/NO questions that can be used to decide whether to include a specific section of the reporting software, or support software to make recommendations.
+To do so, we use the ASK keyword. For example, for the question “Did Elton John have an occupation?” we could ask
+
+ASK WHERE { dbr:Elton_John dbo:occupation ?any }  
+
+The answer would be True (yes). Your assignment is to ask a similar question about Aristotle by using FILTER.
+You must also have your answer as False (no).
+
+Solution:
+ASK WHERE { FILTER NOT EXISTS { dbr:Elton_John dbo:occupation ?any }}
```
Comment (giacomodecolle) [inline Project 4/Ali_Project4:25]: Why is Elton John in the solution again?

File: Project 4/Ali_Project4 [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+WHERE {
+    { 
+      ?s ex:hasFeature wiki:SPARQL . }
+    UNION {
+      ?s schema:mentions wiki:SPARQL . }
+    UNION {
+      ?s schema:keywords "SPARQL" . }
+}
+LIMIT 100
+
```
Comment (giacomodecolle) [inline Project 4/Ali_Project4:43]: I think this works, perhaps you could help the user by saying something more specific about the properties they need to use.  
Also, you use ex:hasFeature but you never mention ex: in the prefixes.
Also, do you think we would achieve the same result if we wrote 
WHERE {
    { 
      ?s ex:hasFeature wiki:SPARQL ;
           schema:mentions wiki:SPARQL ;
          schema:keywords "SPARQL" . 
}

File: Project 4/Ali_Project4 [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+--------------------------
+Ali's Kata 5 question (not working!!!): You can write queries to see which specific class(es) are the most used ones as subject/predicate/object in an ontology. 
+Write a query to find what is the most used subject and object in the OWL dataset.
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+
+SELECT ?class (MAX(?s) AS ?number)
+WHERE {
+    {
+      ?class ?p ?o . }
```
Comment (giacomodecolle) [inline Project 4/Ali_Project4:53]: I think that if you write clauses with three variables you are matching anything in the dataset. Basically, this doesn't seem to be imposing any condition on the graph.
Maybe ?s rdfs:class ?o
And then you can COUNT(?s) and ORDER BY DESC(?s)
or something along these lines.

File: Project 4/Ali_Project4 [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Ali's Kata 4 question: Make a SPARQL query that uses dbpedia to construct a graph of all cities with a population greater that one million.
+
+PREFIX dbo: <http://dbpedia.org/ontology/>
+PREFIX dbp: <http://dbpedia.org/resource/>
+
+CONSTRUCT {
+  ?city rdf:type dbo:City ;
+        dbo:populationTotal ?population ;
+        dbo:abstract ?abstract ;
+        dbo:thumbnail ?thumbnail ;
```
Comment (giacomodecolle) [inline Project 4/Ali_Project4:68]: what do abstract and thumbnail do?

I am not an expert of the CONSTRUCT syntax but this seems to work. Not sure whether it is a level 4 though, sounds more like level 5 to me.

File: Project 4/Ali_Project4 [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    ?occupation rdfs:label ?label .
+    FILTER(LANG(?label) = "en").
+    FILTER(CONTAINS(?label, "Graphic")).
+    FILTER(CONTAINS(?label, "novelist")).
+    FILTER NOT EXISTS {
+      ?person dbo:deathDate ?deathDate .
+    }
+} 
+LIMIT 5
+
```
Comment (giacomodecolle) [inline Project 4/Ali_Project4:100]: I like this one a lot. I think level 5 or 4. Perhaps I am being too heavy on my own level 5s.

**Comment by johnbeve** [comment]
@ahasanzadeh13 This isn't ready for me to review, right? Maybe move the discussion to your repo?

File: Project 4/Ali_Project4 (hunk 1 above)
Comment (peihongx) [reply Project 4/Ali_Project4:25]: Second that.

File: Project 4/Ali_Project4 (hunk 2 above)
Comment (peihongx) [reply Project 4/Ali_Project4:43]: @giacomodecolle Your alternative query requires the same resource (?s) to have a value for 3 different properties at the same time. But maybe some resource has a value only for one of them. @ahasanzadeh13 But if we cannot guarantee that all resources have a value for the property "ex:hasFeature", then we should use OPTIONAL. So our query will look like:

WHERE {
   OPTIONAL 
   {?s ex:hasFeature wiki:SPARQL . }
    OPTIONAL 
   {?s schema:mentions wiki:SPARQL . }
    OPTIONAL 
   {?s schema:keywords "SPARQL" . }
}

File: Project 4/Ali_Project4 [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+}
+LIMIT 100
+
+--------------------------
+Ali's Kata 5 question (not working!!!): You can write queries to see which specific class(es) are the most used ones as subject/predicate/object in an ontology. 
+Write a query to find what is the most used subject and object in the OWL dataset.
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+
+SELECT ?class (MAX(?s) AS ?number)
```
Comment (peihongx) [inline Project 4/Ali_Project4:50]: I suspect whether we can use MAX function directly here. It seems that MAX can be used only if values of the variable ?s it bounds are numbers, but here values of ?s are clearly URIs. So I think that you should first BIND(COUNT(?s) AS ?number) and then use MAX to find the largest value of ?number.

File: Project 4/Ali_Project4 [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+WHERE {
+    {
+      ?class ?p ?o . }
+    UNION {
+      ?s ?p ?class . }
+}
+
+--------------------------
+Ali's Kata 4 question: Make a SPARQL query that uses dbpedia to construct a graph of all cities with a population greater that one million.
```
Comment (peihongx) [inline Project 4/Ali_Project4:59]: I suspect that this is a kata 5. Maybe you can make it harder by adding more sophicated filtering conditions.

File: Project 4/Ali_Project4 [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        dbo:populationTotal ?population ;
+        dbo:abstract ?abstract ;
+        dbo:thumbnail ?thumbnail ;
+        dbo:country ?country .
+  
+  FILTER (?population > 1000000)
+}
+
+--------------------------
+Ali's Kata 4 question: Using FILTER is a way to narrow down your result based on any form of specifity you'd like in SPARQL.
```
Comment (peihongx) [inline Project 4/Ali_Project4:81]: Also, I suspect that this is a kata 5 lol

**Pull request closed without merging**